tests/components/icloud/test_config_flow.py

```diff
@@ -252,6 +266,19 @@ def mock_controller_service_validate_verification_code_failed():
         assert result["errors"] == {CONF_USERNAME: "login"}
 
 
+async def test_no_device(
+    hass: HomeAssistantType, service_authenticated_no_device: MagicMock
+):
+    """Test when we have no devices."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "no_device"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5152307)